R/model.R

```diff
@@ -1360,33 +1395,196 @@ artnum_mf <- function(calendar_quarter, art_number, naomi_mf) {
+
+  # Collate and format data inputs to match outputs with column names:
+  #  "area_id", "sex" , "age_group" , "calendar_quarter", "indicator","mean",
+  # "se" ,"median" , "mode" , "lower" , "upper", "data_type"
+
+  format_inputs_to_outputs <- function(dat, source = "programme_data") {
+
+    indicators <- c("art_current", "art_new", "vl_tested_12mos", "vl_supressed_12mos",
+                    "anc_clients", "anc_known_pos","anc_already_art", "anc_tested",
+                    "anc_tested_pos")
+
+
+    ind <- intersect(colnames(dat), indicators)
+
+    long <- dat %>%
+      dplyr::select(area_id,  sex , age_group, calendar_quarter, data_type, all_of(ind)) %>%
+      tidyr::pivot_longer(cols = all_of(ind),
+                          names_to = "indicator", values_to = "mean") %>%
+      dplyr::mutate(se = NA_real_ ,median = NA_real_ , mode = NA_real_ ,
+                    lower = NA_real_ , upper = NA_real_, source = source)
+
+
+  }
+
+  # Collate and format survey_data
+  input_dat <- rbind(naomi_data$prev_dat$raw_input,
+                     naomi_data$artcov_dat$raw_input,
+                     naomi_data$recent_dat$raw_input,
+                     naomi_data$vls_dat$raw_input) %>%
+    dplyr::mutate(median = NA_real_, mode = NA_real_) %>%
+    dplyr::select(area_id, sex, age_group, calendar_quarter = survey_mid_calendar_quarter,
+                  indicator, data_type, mean = estimate, se = std_error,lower = ci_lower,
+                  upper = ci_upper, median, mode, source = survey_id)
+
+
+  # Collate and format ART data
+  art_dat <- rbind(naomi_data$artnum_t1_dat$raw_input,
+                   naomi_data$artnum_t2_dat$raw_input)
+
+  if(!is.null(art_dat) && nrow(art_dat) > 1) {
+    art_long <- format_inputs_to_outputs(art_dat)
+
+    input_dat <- rbind(input_dat, art_long)
+  }
+
+
+  # Collate and format ANC data
+
+  anc_keep_cols <- c("area_id", "age_group", "sex" ,"year", "anc_clients", "anc_known_pos",
+                 "anc_already_art", "anc_tested","anc_tested_pos", "anc_clients", "data_type")
+
+
+
+  anc_artcov_dat <- rbind(naomi_data$anc_artcov_t1_dat$raw_input,
+                           naomi_data$anc_artcov_t2_dat$raw_input)
+  anc_prev_dat <- rbind(naomi_data$anc_prev_t1_dat$raw_input,
+                        naomi_data$anc_prev_t2_dat$raw_input)
+
+  silence_cols <- intersect(colnames(anc_artcov_dat),
+                            colnames(anc_prev_dat))
+
+  anc_dat <- dplyr::left_join(anc_artcov_dat %>%
+                     dplyr::select(intersect(colnames(anc_artcov_dat), anc_keep_cols)),
+                     anc_prev_dat %>%
+                     dplyr::select(intersect(colnames(anc_prev_dat), anc_keep_cols)),
+                     by = silence_cols)
+
+
+  if(!is.null(anc_dat) && nrow(anc_dat) > 1) {
+
+    anc_dat <- anc_dat %>%
+      dplyr::distinct() %>%
+      dplyr::mutate(calendar_quarter = paste0("CY", year, "Q4"), sex = "female")
+
+    anc_long <- format_inputs_to_outputs(anc_dat)
```

Comment:
This table is missing `anc_prevalence` and `anc_art_coverage` indicators.  (Not sure where in code that needs to be added; leaving comment here.